src/widgets/Updater.cpp

```diff
@@ -20,8 +20,29 @@
 *//******************************************************************/
 
 #include "Updater.h"
+#include <wx/font.h>
+#include <wx/dc.h>
 
-Updater::TickSizes::TickSizes(double UPP, int orientation, Ruler::RulerFormat format, bool log)
+RulerStruct::RulerStruct()
+{
+   mMin = mHiddenMin = 0.0;
```

Comment:
The second commit introduces `RulerStruct` and this constructor.

This constructor has many assignments in it.

It's better C++ style (since 1998) to prefer member initialization where possible over assignments in the constructor body, for instance:

```
RulerStruct::RulerStruct()
   : mMin(0.0)
{
```

But since C++11, it's even better to prefer the "uniform initialization" syntax (curly braces) where possible:

```
RulerStruct::RulerStruct()
   : mMin{ 0.0 }
{
```

-- among other things, this disallows implicit narrowing convertions (as from float to integer).

But, also in C++11, and better still where it applies:  use in-class initialization.  That is, in Updater.h, declare the member as:

```
double mMin = 0.0;
```

or better

```
double mMin{ 0.0 };
```

or, even:

```
double mMin{};
```

Because 0 is the default value for all kinds of number types.

Any of those is better than
```
double mMin;
```

which implies no initialization.

When I review a class to understand it better, I want to start with the header file.  If I see in-class initializers, I don't have to read the constructor definitions to have confidence that members are never left uninitialized by mistake.  Rather I know that they default reasonably.  And then when I do read the constructor, it's a shorter constructor.

Only when there is no reasonable, argument-independent default for a member -- then, leave it without an initializer.  But then, I know that I must read the constructor body to understand fully what the member is for.